### ISSUE TITLE
README.mc needs instructions for how to manually update the lib for developers

### ISSUE THREAD

**Issue by ensonic** [rhempel/ev3dev-lang-python#287]
Seriously, either keep it simple or tell people how to do it. I've tried:

```
# on the ev3
mkdir ev3dev
# on development laptop:
<make changes>
scp ev3dev/*.py robot@ev3dev.local:ev3dev/
# on ev3
sudo mv *.py /usr/lib/python3/dist-packages/ev3dev/
```

but after that I only get python errors (e.g. instantiating motors fails).

**Comment by ddemidov** [comment]
What I do in order to test things out, is
```
ssh robot@ev3dev
git clone https://github.com/rhempel/ev3dev-lang-python
cd ./ev3dev-lang-python
```
Now, you can do your changes to the local copy. When you do
```py
import ev3dev.ev3 as ev3
```
while in the `ev3dev-lang-python` folder, it should import from local folder instead of from system-installed one.

**Comment by ensonic** [comment]
Unfortunately this won't work for installed stuff. The openrobertalab is a package with a service run by systemd.

**Comment by ddemidov** [comment]
Well, the installed stuff is installed with apt-get, so I guess the only correct way to test things would be to uninstall the system package first, then run `./setup.py install` from the cloned repo.

**Comment by ensonic** [comment]
Nah, just need to get the right files to the right locations. I'll check the layout of the src tree and the installed files.

**Comment by WasabiFan** [comment]
It sounds like you're looking for the equivalent of [`npm link`](https://docs.npmjs.com/cli/link) from the JavaScript world... I can't seem to find anything similar in Python package managers though.

**Comment by dwalton76** [comment]
You can do
```
ssh robot@ev3dev
git clone https://github.com/rhempel/ev3dev-lang-python
cd ./ev3dev-lang-python
sudo python3 setup.py install
```

Then whenever you make changes run the "sudo python3 setup.py install" again

**Comment by dwalton76** [comment]
@ensonic can you confirm that does the trick for openrobertalab

**Comment by ensonic** [comment]
@dwalton76 if I follow your instructions, it will be installed to /usr/local/lib/python3.4/dist-packages/ where as the debian package is in /usr/lib/python3/dist-packages/. That usually is a recipe for disaster (especially so since setup.py cannot easily uninstall stuff).

**Comment by dwalton76** [comment]
Right you do end up with two copies which is not ideal but I've never run into issues with it since the copy in /usr/local/lib/python3.4/dist-packages/ is found first via your PYTHON_PATH.  I just rm -rf the /usr/local/lib/python3.4/dist-packages/ copy if I want to go back to using the version installed via the deb.  Most of the time I just end up using the copy installed via setup.py though as I'm continuing to do dev on it.

You could always uninstall the copy from the debian package to avoid having two copies.

**Comment by ensonic** [comment]
I can't uninstall the python library, since that will install packages depending on it. I'll send a PR for the docs with the gist of this thread.

**Comment by WasabiFan** [comment]
I believe this is the currently recommented list of options:
- Update `PYTHONPATH`
- Do development from the directory of a local copy of the library (current working directory is searched for modules)

Feel free to update this with more. I'm not going to close this immediately as it would be nice to add complete instructions to the README.

**Comment by dlech** [comment]
no underscore in `PYTHONPATH` environment variable name.

**Comment by WasabiFan** [comment]
Updated. I could've sworn there was one 🤔

**Comment by WasabiFan** [comment]
It looks like this was mostly resolved so I'm going to close the issue. Please reopen or submit a PR if there is a change that's still needed.
